src/renderers/shared/reconciler/instantiateReactComponent.js

```diff
@@ -122,6 +122,7 @@ function instantiateReactComponent(node) {
   if (__DEV__) {
     instance._isOwnerNecessary = false;
     instance._warnedAboutRefsInRender = false;
+    instance._isServerSideRendered = false;
```

Comment:
It is a little weird for this to be `false` (a boolean) or an object.  Maybe rename the variable and use `null` instead of `false`?